Wikipedia/Code/WMFAppViewController.m

```diff
@@ -362,6 +362,7 @@ - (BOOL)processUserActivity:(NSUserActivity*)activity {
             [[UIViewController wmf_sharedSearchViewController] setSearchTerm:[activity wmf_searchTerm]];
             break;
         case WMFUserActivityTypeArticle: {
+            [self dismissViewControllerAnimated:NO completion:NULL];
```

Comment:
Looks like we are dismissing in all cases - mind DRYing it up and putting the dismiss before the case statement?